rust_src/src/multibyte.rs

```diff
@@ -950,3 +1030,43 @@ pub fn char_to_byte8(c: Codepoint) -> u8 {
 pub const fn single_byte_charp(c: Codepoint) -> bool {
     c < 0x100
 }
+
+/// Wrapper function for find_composition in src/composite.c
+pub fn find_composition(
+    pos: usize,
+    limit: Option<usize>,
+    object: LispObject,
+) -> Option<(usize, usize, LispObject)> {
+    let pos = pos as isize;
+    let limit = limit.map_or(-1, |l| l as isize);
+    let (mut start, mut end, mut prop) = (0, 0, Qnil);
+    unsafe {
+        if c_find_composition(pos, limit, &mut start, &mut end, &mut prop, object) {
+            Some((start as usize, end as usize, prop))
+        } else {
+            None
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn lisp_string_width(
+    string: LispObject,
+    precision: isize,
+    nchars: *mut isize,
+    nbytes: *mut isize,
+) -> isize {
+    let precision = if precision < 0 {
```

Comment:
An option here is using `TryInto` couple with `usize`.

Reply:
Done, that looks much nicer.